core/src/main/java/org/elasticsearch/index/query/SimpleQueryStringBuilder.java

```diff
@@ -438,7 +435,7 @@ public boolean doEquals(SimpleQueryStringBuilder other) {
         return Objects.equals(fieldsAndWeights, other.fieldsAndWeights) && Objects.equals(analyzer, other.analyzer)
                 && Objects.equals(defaultOperator, other.defaultOperator) && Objects.equals(queryText, other.queryText)
                 && Objects.equals(queryName, other.queryName) && Objects.equals(minimumShouldMatch, other.minimumShouldMatch)
-                && Objects.equals(settings, other.settings);
+                && Objects.equals(settings, other.settings) && (flags == other.flags);
```

Comment:
What isn't equal isn't equal. Hash and equals are consistent, hard for any test to see if its okay or not unless we flip every allowed field and check that afterwards things are not equal, I guess.